test/hotspot/jtreg/compiler/loopopts/superword/TestAliasingCastP2XCtrl.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test id=all-flags
+ * @bug 8366490
+ * @summary Test that we set the ctrl of CastP2X when generating
+ *          the aliasing runtime check, preventing the CastP2X
+ *          from floating over a SafePoint that could move the oop,
+ *          and render the cast value stale.
+ *
+ * @requires vm.gc == "G1" | vm.gc == "null"
+ * @run main/othervm
+ *      -XX:+IgnoreUnrecognizedVMOptions
+ *      -XX:CompileCommand=compileonly,*TestAliasingCastP2XCtrl::test
+ *      -XX:CompileCommand=dontinline,*TestAliasingCastP2XCtrl::allocateArrays
+ *      -XX:-TieredCompilation
+ *      -Xbatch
+ *      -XX:+UseG1GC
+ *      -XX:+UnlockDiagnosticVMOptions -XX:+StressGCM
+ *      compiler.loopopts.superword.TestAliasingCastP2XCtrl
+ */
+
+/*
+ * @test id=fewer-flags
+ * @bug 8366490
+ * @run main/othervm
+ *      -XX:+IgnoreUnrecognizedVMOptions
+ *      -XX:CompileCommand=compileonly,*TestAliasingCastP2XCtrl::test
+ *      -XX:CompileCommand=dontinline,*TestAliasingCastP2XCtrl::allocateArrays
+ *      -Xbatch
+ *      -XX:+UnlockDiagnosticVMOptions -XX:+StressGCM
+ *      compiler.loopopts.superword.TestAliasingCastP2XCtrl
+ */
+
+/*
+ * @test id=vanilla
+ * @bug 8366490
+ * @run driver compiler.loopopts.superword.TestAliasingCastP2XCtrl
```

Comment:
Should be `main` to allow to run with passed in flags
```suggestion
 * @run main compiler.loopopts.superword.TestAliasingCastP2XCtrl
```